File: src/main/java/com/zoarial/iot/threads/udp/DatagramPacketHelper.java
package com.zoarial.iot.threads.udp;

import com.zoarial.PrintBaseClass;

import java.io.*;
import java.net.DatagramPacket;
import java.util.UUID;

public class DatagramPacketHelper extends PrintBaseClass {
    public final BufferedInputStream rawIn;
    public final DataInputStream in;

    public DatagramPacketHelper(DatagramPacket dp) {
        super("Datagram Packet Helper");
        rawIn = new BufferedInputStream(new ByteArrayInputStream(dp.getData()));
        in = new DataInputStream(rawIn);
    }

    public String readString() throws IOException {
        StringBuilder str = new StringBuilder();
        byte b = in.readByte();
        while(b != 0) {
            str.append((char)b);
            b = in.readByte();
        }
        return str.toString();
    }

    public UUID readUUID() throws IOException {
        return new UUID(in.readLong(), in.readLong());
    }

    public int readInt() throws IOException {
        return in.readInt();
    }

    public long readLong() throws IOException {
        return in.readLong();
    }

    public byte readByte() throws IOException {
        return in.readByte();
    }

    public byte[] readBytes(int i) throws IOException {
        return in.readNBytes(i);
    }
}

File: src/main/java/com/zoarial/iot/threads/udp/DatagramPacketHelper.kt
package com.zoarial.iot.threads.udp

import com.zoarial.PrintBaseClass
import java.io.BufferedInputStream
import java.io.ByteArrayInputStream
import java.io.DataInputStream
import java.io.IOException
import java.net.DatagramPacket
import java.util.*

class DatagramPacketHelper(dp: DatagramPacket) : PrintBaseClass("Datagram Packet Helper") {
    val rawIn: BufferedInputStream
    val `in`: DataInputStream

    init {
        rawIn = BufferedInputStream(ByteArrayInputStream(dp.data))
        `in` = DataInputStream(rawIn)
    }

    @Throws(IOException::class)
    fun readString(): String {
        val str = StringBuilder()
        var b = `in`.readByte()
        while (b.toInt() != 0) {
            str.append(Char(b.toUShort()))
            b = `in`.readByte()
        }
        return str.toString()
    }

    @Throws(IOException::class)
    fun readUUID(): UUID {
        return UUID(`in`.readLong(), `in`.readLong())
    }

    @Throws(IOException::class)
    fun readInt(): Int {
        return `in`.readInt()
    }

    @Throws(IOException::class)
    fun readLong(): Long {
        return `in`.readLong()
    }

    @Throws(IOException::class)
    fun readByte(): Byte {
        return `in`.readByte()
    }

    @Throws(IOException::class)
    fun readBytes(i: Int): ByteArray {
        return `in`.readNBytes(i)
    }
}